src/apps/xpubs/xpubs.py

```diff
@@ -69,10 +73,59 @@ async def menu(self, show_screen, show_all=False):
             if der is not None:
                 await self.show_xpub(der, show_screen)
                 return True
+        elif menuitem == 3:
+            format = await self.save_menu(show_screen)
+            if format is not False:
+
+                fingerprint = hexlify(self.keystore.fingerprint).decode()
+                extension = "json"
+                coin = net["bip32"]
+
+                derivations = [
+                    ('bip49', "p2wpkh", "m/49'/%d'/%d'" % (coin, self.account)),
+                    ('bip84', "p2sh-p2wpkh", "m/84'/%d'/%d'" % (coin, self.account)),
+                    ('bip44', "p2pkh", "m/44'/%d'/%d'" % (coin, self.account)),
+                    ('bip48_2', "p2wsh", "m/48'/%d'/%d'/2'" % (coin, self.account)),
+                    ('bip48_1', "p2sh-p2wsh", "m/48'/%d'/%d'/1'" % (coin, self.account)),
+                ]
+
+                if format == self.export_specter_diy:
+                    filedata = ""
+                    for der in derivations:
+                        filedata += "[%s/%s]%s\n" % (fingerprint, der[2], self.calculate_xpub(der[2]))
+                    extension = "txt"
+                else:
+                    m = self.keystore.get_xpub("m")
+                    data = {
+                        "xpub": m.to_base58(NETWORKS[self.network]["xpub"]),
+                        "xfp": fingerprint,
+                        "account": self.account,
+                        "chain": "BTC" if self.network == "main" else "XTN"
+                    }
+
+                    for der in derivations:
+                        data[der[0]] = {
+                            "name": der[1],
+                            "deriv": der[2],
+                            "xpub": self.calculate_xpub(der[2]),
+                            "_pub": self.calculate_xpub(der[2], True),
```

Comment:
Calculating xpubs is an expensive function (it does some EC math in `keystore.get_xpub`),
so better to call `keystore.get_xpub` once and then serialize it to xpub and slip132.
I'd suggest deriving the key in this function once per derivation path, and then serializing it with proper version.

Reply:
Will rewrite